integration/build/build_test.go

```diff
@@ -0,0 +1,101 @@
+package build
+
+import (
+	"archive/tar"
+	"bytes"
+	"context"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/integration/util/request"
+)
+
+func TestBuildFailuresAndIntermediateContainers(t *testing.T) {
```

Comment:
Maybe `TestBuildWithRemoveAndForceRemove` or something like that? Intermediate containers seem like more of the test result than the test condition. And this tests both failures and non-failures, so failures doesn't seem relevant to the name.